pkg/realizer/runnable/realizer.go

```diff
@@ -147,40 +149,39 @@ func (p *runnableRealizer) Realize(ctx context.Context, runnable *v1alpha1.Runna
 	return stampedObject, outputs, nil
 }
 
-func resolveSelector(ctx context.Context, selector *v1alpha1.ResourceSelector, repository repository.Repository, namespace string) (map[string]interface{}, error) {
+func (r *runnableRealizer) resolveSelector(ctx context.Context, selector *v1alpha1.ResourceSelector, repository repository.Repository, discoveryClient discovery.DiscoveryInterface, namespace string) (map[string]interface{}, error) {
 	log := logr.FromContextOrDiscard(ctx)
 
 	if selector == nil {
 		return nil, nil
 	}
 
-	results, err := repository.ListUnstructured(ctx, schema.FromAPIVersionAndKind(selector.Resource.APIVersion, selector.Resource.Kind), namespace, selector.MatchingLabels)
+	apiResourceList, err := discoveryClient.ServerResourcesForGroupVersion(selector.Resource.APIVersion)
```

Comment:
do you know if there's any impact performance-wise with having this being run on every `resolveSelector` call? I'm not very familiar with the internals of the discovery client, but, IIRC, `kubectl` makes use of a cached version of it that persists results on disk to make things fast 🤔

Reply:
switched to the cached client, as you suggested below.